### PR TITLE
Feature add hreflang x-default tag to sitemap

### LINKED ISSUE #161: Add an additional link tag with hreflang x-default to improve locale detection fallback
Hello, thank you for this really useful plugin !

### Summary

Add a `link` tag with `hreflang="x-default"`

### Why is it needed?

Ahref SEO software rises advice to enable link tag with hreflang x-default value for default language fallback.
Google also mentions it: https://developers.google.com/search/blog/2023/05/x-default?hl=en

It is not necessary nor seems to penalise seo. But I would like to know your opinion on it?

### Suggested solution(s)

Add the additional `link` tag with `hreflang="x-default"` value from default locale pathname.
I could contribute if needed 😃

### PR DESCRIPTION
**Pull request by RomainGueffier**
### What does it do?

Add an additional link tag with hreflang x-default to improve locale detection fallback.
Add in settings form a select to save user preferences with 3 options:
- Disabled
- Default language pathname
- Other
When last option is selected, an extra text input field shows up to put a custom url.
Add translations for English and French
Extend default Test collection to add `slug` field in order to have better use cases to try feature.

### Why is it needed?

Improve SEO.

### How to test it?

Run strapi with plugin installed. Open Sitemap plugin settings.

Outputs:

1. Disabled

disabled (default)

2. Default Locale

default-locale (bundles)

3. Other

Other (custom)

### Related issue(s)/PR(s)

Closes #161 

### Tasks:

- [x] add more translations ?
- [x] can't find how to update docs via `yarn docs:generate` mentioned in CONTRIBUTING.md.
- [x] improve labels and descriptions to help users understand what it does and SEO risks.

> ⚠️ `yarn develop / yarn playground:develop` did not update admin UI on save, whereas server was ok. Is it normal ? I needed to restart playground on each UI changes.

### CODE REVIEW

File: server/services/core.js [hunk 1]
```diff
@@ -10,6 +10,32 @@ const { isEmpty } = require('lodash');
...
+ *
+ * @returns {object | undefined} The default language link.
+ */
+const getDefaultLanguageLink = async (config, links) => {
+  let defaultLink;
+  const { getDefaultLocale } = strapi.plugin('i18n').service('locales');
+  const defaultLocale = await getDefaultLocale();
+
+  if (config.defaultLanguageUrlType === 'default-locale') {
+    // find url with default locale in generated bundle
```
Comment (boazpoolman) [inline server/services/core.js:27]: We only have to fetch the defaultLocale if the setting is actually set to 'default-locale'. Saves on some unnecessary queries.
```suggestion

  if (config.defaultLanguageUrlType === 'default-locale') {
    const { getDefaultLocale } = strapi.plugin('i18n').service('locales');
    const defaultLocale = await getDefaultLocale();
  
    // find url with default locale in generated bundle
```

File: admin/src/tabs/Settings/index.jsx [hunk 2]
```diff
@@ -88,6 +97,41 @@ const Settings = () => {
...
+        >
+          <SingleSelectOption value="">
+            {formatMessage({ id: 'sitemap.Settings.Field.DefaultLanguageUrlType.Option.Disabled', defaultMessage: 'Disabled' })}
+          </SingleSelectOption>
+          <SingleSelectOption value="default-locale">
+            {formatMessage({ id: 'sitemap.Settings.Field.DefaultLanguageUrlType.Option.DefaultLocale', defaultMessage: 'Default language URL of bundles (generated from default locale URL)' })}
+          </SingleSelectOption>
+          <SingleSelectOption value="other">
+            {formatMessage({ id: 'sitemap.Settings.Field.DefaultLanguageUrlType.Option.Other', defaultMessage: 'Other' })}
+          </SingleSelectOption>
```
Comment (boazpoolman) [inline admin/src/tabs/Settings/index.jsx:119]: Could you abstract these `defaultLanguageUrlType` values to the `admin/config/constants.js` file?

Something like:

```
export const DEFAULT_LANGUAGE_URL_TYPE_DEFAULT_LOCALE = 'default-locale';
export const DEFAULT_LANGUAGE_URL_TYPE_OTHER = 'other';
```

Then everywhere these values are referenced you can use those constants.

**Review by boazpoolman — CHANGES_REQUESTED** [review]
Thanks for this PR @RomainGueffier.
Functionally it works perfect!

I just have a couple remarks on your code.
If you could implement those changes I'll get this merged and released asap :)

**Comment by boazpoolman** [comment]
> ⚠️ yarn develop / yarn playground:develop did not update admin UI on save, whereas server was ok. Is it normal ? I needed to restart playground on each UI changes.

You can run `yarn playground:develop --watch-admin` to make sure the admin also updates on save.

> can't find how to update docs via yarn docs:generate mentioned in CONTRIBUTING.md.

Sorry. Those `yarn docs:` commands are outdated and should be removed from the `package.json`

File: admin/src/translations/fr.json [hunk 3]
```diff
@@ -1 +1,9 @@
...
\ No newline at end of file
+{
+  "Settings.Field.DefaultLanguageUrlType.Label": "Type d'URL de la langue par défaut.",
+  "Settings.Field.DefaultLanguageUrlType.Description": "Génère une balise link et attribut hreflang=x-default avec l'URL de votre choix.",
+  "Settings.Field.DefaultLanguageUrl.Label": "URL de la langue par défaut personnalisée.",
+  "Settings.Field.DefaultLanguageUrl.Description": "Ex. URL de la page de sélection de langue.",
+  "Settings.Field.DefaultLanguageUrlType.Option.Disabled": "Désactivé",
+  "Settings.Field.DefaultLanguageUrlType.Option.DefaultLocale": "URL par défaut des routes (généré à partir de l'URL de la locale par défaut)",
+  "Settings.Field.DefaultLanguageUrlType.Option.Other": "Autre"
+}
```
[the remark this answers is not in the document]
Comment (RomainGueffier) [reply admin/src/translations/fr.json:9]: hehe I was going to offer my translation's services in another PR 😄 but if you want I can do it here

**Pull request merged**